doc/litespeed-configuration.rst

```diff
@@ -0,0 +1,137 @@
+.. _litespeed configuration:
+
+LiteSpeed Configuration
+-----------------------
+
+Below you will find detailed LiteSpeed configuration recommendations for the
+features provided by this library.
+
+Preamble
+~~~~~~~~
+
+First of all, let's get one thing straight here: You'll find a lot of documentation
+and noise around LiteSpeed cache on the Internet, mostly involving plugins, specifically the
+Wordpress one. You **don't** need any plugin to benefit from LiteSpeed cache!
+As long as you follow the HTTP specification regarding the caching headers, you can use it as
+a general reverse proxy just like NGINX or Varnish.
+
+LiteSpeed comes in two different variants:
+
+* OpenLiteSpeed (OLS) - the open source product with less features
+* LiteSpeed Web Server (LSWS) - the enterprise version with more features and professional support
+
+The caching module implementations are different and thus have to be configured differently but they support the
+same set of features (except for OLS not supporting Edge Side Includes (ESI)).
+
+So before you start configuring the server, make sure you know which version of LiteSpeed you are using.
+
+.. note::
+
+    Any LiteSpeed setup works in a single node web server environment only. If you are targeting a multi
+    node setup you might want to consider switching to :ref:`Varnish <varnish configuration>` which has excellent
+    support for this already built-in in this library.
+
+
+Configuring OpenLiteSpeed
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+OLS does not support different caching settings depending on ``.htaccess`` settings and different paths.
+If you need that, you have to go with LSWS instead.
+Thus, OLS has to be configured as follows on server or vHost level::
+
+    module cache {
+      # This enables the public cache
+      enableCache                      1
+
+      # This disables the private cache
+      enablePrivateCache               0
+
+      # This enables the public cache
+      checkPublicCache                 1
+
+      # This disables the private cache
+      checkPrivateCache                0
+
+      # Also consider the query string in caches
+      qsCache                          1
+
+      # Enable checking for a cached entry if there's a cookie on the request
+      reqCookieCache                   1
```

Comment:
should we add a line here to mention that you need to enable user context caching if you set this to 1 and otherwise risk mixing up your cache? or maybe we should default 0 and then have a commented line with 1 and explain to use that with user context caching?

Reply:
I would prefer not to mention it because I haven't tested if that would really work.

Reply:
without user context, we should really not enable caching when there are cookies, otherwise you get cache mixups when there are session cookies

Reply:
We're not caching. We're serving entries from the cache even if you're logged in. Depends on your use case but if handled correctly, that's a valid use case. E.g. all your public pages can still be served from the cache even if one of your visitors is logged in. However, you have to make sure that these "public" pages are really public or if they contain parts that differ when one is logged in, they are served via ESI. 
So really it depends on the use case and I agree, you have to be pretty fit in terms of caching to understand it all. But on the other hand, we cannot write an essay here. Do you think we should disable it in the docs and just mention that one might want to enable this depending on the use case?

Reply:
so the use case would be that you have some pages that do not depend on the session in at all, and you configure the cache to allow caching of everything and the backend is supposed to cache-control with that in mind? 

if i got that correctly, none of our other documentations suggest this strategy. i would not put that into such an example with a list of configuration options that people might blindly copy-paste and end up with a very risky system if they are not aware what they did.

if we want to cover this use case in the doc, it should be a new section, explaining for each cache how to configure the behaviour and clearly warning of the consequences if the backend does not set cache instructions correctly.

Reply:
No, I think you still didn't get the use case right 😄
Let's say we have an `/about-us.html` which is public. The page has a private section where you can log in so when you visit `/about-us.html` you potentially have a cookie set.
So in this case I don't want OLS to ignore the cache for `/about-us.html` because I know it's not going to be different.
But yes, I agree you have to know every response of your app and you have to always keep that in mind. So I'm probably going to change this for the docs, I can still enable it for myself 😄

Reply:
hm, how is that different from what i described? you are (maybe) logged in and go to a page and we tell the cache that if the backend wants that page cached, it should cache it even if there are credentials.

as said, it would be ok to cover that use case in the docs (as kindof a simplified version of user context that does not need the hash request). but we should not have it here as a "default" configuration, its too dangerous for that.

Reply:
Agreed.